Solves issues related to transitionArchetype

TheyCallMeRyan opened the pull request:
## Proposed changes

If the oldArchetype is the same as the new archetype then don't do any transition

## Related issues

See: https://github.com/matter-ecs/matter/issues/157

## Additional comments

TheyCallMeRyan committed:
Prevents transitions from an archetype to itself

memorycode commented:
Thanks for the PR. Are you able to add a test to make sure that this fixes the issue?

TheyCallMeRyan committed:
Add function removeFromArchetype

Splits some of the logic from transitionArchetype into a separate function removeFromArchetype that can be utilized by both transitionArchetype and executeDespawn. 

Removes executeDespawn's dependency on transitionArchetype, solving the bug presented in #164.

TheyCallMeRyan commented:
Added another commit resolving the issue in #164 as well

LastTalon commented:
Can we please edit the title to reflect the changes that are happening and link appropriately with closing keywords before merging this?
